src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -219,31 +228,49 @@ abstract class AbstractCheckTask(
             .subscribe { _ -> end("Timeout is exited") }
 
-        messageStream.observeOn(scheduler) // Defined scheduler to execution in one thread to avoid race-condition.
-            .doFinally(this::taskFinished) // will be executed if the source is complete or an error received or the timeout is exited.
-
-            // All sources above will be disposed on this scheduler.
-            //
-            // This method should be called as closer as possible
-            // to the actual dispose that you want to execute on this scheduler
-            // because other operations are executed on the same single-thread scheduler.
-            //
-            // If we move [Observable#unsubscribeOn] after them, they won't be disposed until the scheduler is free.
-            // In the worst-case scenario, it might never happen.
-            .unsubscribeOn(scheduler)
-            .continueObserve(sessionKey, sequence)
-            .doOnNext {
-                handledMessageCounter++
-
-                with(it.metadata.id) {
-                    rootEvent.messageID(this)
-                }
-            }
-            .mapToMessageContainer()
-            .taskPipeline()
-            .subscribe(this)
+        try {
+            messageStream.observeOn(scheduler) // Defined scheduler to execution in one thread to avoid race-condition.
+                    .doFinally(this::taskFinished) // will be executed if the source is complete or an error received or the timeout is exited.
+
+                    // All sources above will be disposed on this scheduler.
+                    //
+                    // This method should be called as closer as possible
+                    // to the actual dispose that you want to execute on this scheduler
+                    // because other operations are executed on the same single-thread scheduler.
+                    //
+                    // If we move [Observable#unsubscribeOn] after them, they won't be disposed until the scheduler is free.
+                    // In the worst-case scenario, it might never happen.
+                    .unsubscribeOn(scheduler)
+                    .continueObserve(sessionKey, sequence)
+                    .doOnNext {
+                        handledMessageCounter++
+
+                        with(it.metadata.id) {
+                            rootEvent.messageID(this)
+                        }
+                    }
+                    .mapToMessageContainer()
+                    .taskPipeline()
+                    .subscribe(this)
+        } catch (exception: Exception) {
+            val initialException = getInitialException(exception)
+            rootEvent.addSubEventWithSamePeriod()
+                    .name("An error occurred while executing rule")
+                    .type("internalError")
+                    .status(FAILED)
+                    .bodyData(EventUtils.createMessageBean(initialException.message))
```

Comment:
Please attach the source exception via the Event.exception(exception, true) call